### PR TITLE
Added Google Analytics

### PR DESCRIPTION
**Pull request by mmmavis**
Related to #325 

### Page visits
- [x] Each page visit is being tracked at pageviews

### Events:

- [x] Share Intent – User taps `Hyperlink icon Button` to reveal entry URL
- [x] Visit – User taps `Visit button` on an entry's detailed view
- [x] Get Involved – User taps `Get involved Link` on an entry's detailed view
- [x] Fav – User taps `Heart button` to Fav an entry
- [x] Unfav – User taps `Heart button` to unfav an entry
- [x] Add new intent – Someone taps big plus button
- [x] Nav links clicks (including issues links)
- [x] Entry card - title clicks
- [x] Entry card - thumbnail clicks
- [x] Entry card - "Read more" clicks
- [x] Search – keyword entered(searched)

---

❗️ 
- [ ] Before merging this PR, file a followup ticket to update the GA UA code used here.

### CODE REVIEW

File: components/selector-link/selector-link.jsx [hunk 1]
```diff
@@ -1,10 +0,0 @@
-import React from 'react';
```
Comment (mmmavis) [inline components/selector-link/selector-link.jsx]: this has been renamed to `nav-link.jsx`

File: main.jsx [hunk 2]
```diff
@@ -1,8 +1,24 @@
...
+  // https://developers.google.com/analytics/devguides/collection/analyticsjs/field-reference#location
+  ReactGA.set({ location: window.location.href });
+
+  ReactGA.pageview(`${window.location.pathname}/${window.location.search}`);
+}
+
 ReactDOM.render((
-  <Router routes={routes} history={browserHistory} />
+  <Router routes={routes} history={browserHistory} onUpdate={logPageView} />
 ), document.getElementById(`app`));
```
Comment (mmmavis) [inline main.jsx]: https://github.com/react-ga/react-ga#with-npm

**Comment by mmmavis** [comment]
@alanmoo can you help review this one? I'm using a testing GA UA code here. I will file a DevOps ticket to get an official one later - I just don't want it to block the work here (not sure if we need to wait till we have an official production domain before we can get a UA code).

To test this PR

1. Go to [the GA Dashboard of the testing property that I created](https://analytics.google.com/analytics/web/?pli=1#realtime/rt-content/a91389561w135470868p139643824/%3Fmetric.type%3D1/).
2. Check out this PR and run Pulse locally.
3. Navigate around the site and see if you do see GA events and pageview coming in the Dashboard. (Dashboard -> Real Time -> Behavior(to see page viewes) and Events (to see events)

File: components/project-card/project-card.jsx [hunk 3]
```diff
@@ -67,10 +89,20 @@ export default React.createClass({
     }
   },
   bookmarkProject(bookmarks) {
+    ReactGA.event({
+      category: `${GA_EVENT_PREFIX}Bookmark Button`,
+      action: `Bookmarked`,
+      label: `${this.props.id} - ${this.props.title}`
```
Comment (alanmoo) [inline components/project-card/project-card.jsx]: This label gets repeated a lot, can you extract it out as a variable at the top?

File: components/project-card/project-card.jsx [hunk 4]
```diff
@@ -88,7 +120,34 @@ export default React.createClass({
...
   },
+  handleThumbnailClick() {
+    ReactGA.event({
+      category: `${GA_EVENT_PREFIX}thumbnail`,
+      action: `Clicked`,
+      label: `${this.props.id} - ${this.props.title}`
+    });
+  },
+  handleTitleClick() {
+    console.log(`GA_EVENT_PREFIX`, GA_EVENT_PREFIX);
```
Comment (alanmoo) [inline components/project-card/project-card.jsx]: Remove

Comment (mmmavis) [inline components/project-card/project-card.jsx]: 😂

**Comment by mmmavis** [comment]
@alanmoo I cleaned up the code a bit. 👀 ?

**Pull request merged**